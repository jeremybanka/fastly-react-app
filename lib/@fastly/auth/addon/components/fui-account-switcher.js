import { OtpSetupError, PasswordRotateError } from '@fastly/auth/utils/errors'
import { set, setProperties } from '@ember/object'
import Component from '@ember/component'
import hbs from 'htmlbars-inline-precompile'
import { inject as service } from '@ember/service'

export default Component.extend({
  // services
  session: service(),
  store: service(),

  // Internal Properties
  // ---------------------------------------------------------------------------
  accounts: null,
  tagName: '',
  isRequestRunning: false,
  loading: false,
  error: null,

  // Methods
  // ---------------------------------------------------------------------------
  async fetchAccounts() {
    setProperties(this, {
      loading: true,
      didError: false,
    })

    const user = this.session.user

    const accounts = await this.store
      .query('account', { userId: user.id })
      .catch((err) => {
        console.error(err)
        set(this, 'error', 'fetchError')
      })
      .finally(() => {
        set(this, 'loading', false)
      })

    set(this, 'accounts', accounts)
  },
  /**
   * Switch the current session to the chosen account,
   * bailing early if the chosen account is the currentSession
   *
   * @param {Object} account
   * @param {string} account.id
   * @param {boolean} account.currentSession
   */
  handleChooseAccount({ id, currentSession }) {
    if (currentSession) return

    setProperties(this, {
      isRequestRunning: true,
      didError: false,
    })

    return this.session
      .switchAccount(id)
      .catch((err) => {
        if (err instanceof OtpSetupError) {
          set(this, 'error', 'otpSetupError')
        } else if (err instanceof PasswordRotateError) {
          set(this, 'error', 'passwordRotateError')
        } else {
          set(this, 'error', 'authError')
        }
      })
      .finally(() => {
        if (!this.isDestroyed && !this.isDestroying) {
          set(this, 'isRequestRunning', false)
        }
      })
  },

  handleKeyUp(account) {
    const ENTER_KEY_CODE = 13
    if (event.keyCode === ENTER_KEY_CODE) {
      this.handleChooseAccount(account)
    }
  },

  // Ember Hooks
  // ---------------------------------------------------------------------------
  init() {
    this.fetchAccounts()
    this._super(...arguments)
  },

  // Template
  // ---------------------------------------------------------------------------

  layout: hbs`
    <h1 class='fui-auth-screen__header'>
      Switch Accounts
    </h1>

    <p class='margin-bottom--m padding-x--m  text-align--left'>
      Switch to another account linked to {{session.user.login}}.
    </p>

    {{#if (eq error 'authError')}}
      <FuiAuthError />
    {{else if (eq error 'fetchError')}}
      <FuiAuthError
        @errorText='A problem occured while fetching accounts. Try again or'
      />
    {{else if (eq error 'otpSetupError')}}
      <FuiAuthError
        @errorText='2FA is required. To enable this now, visit your'
        @linkText='account settings'
        @link='/account/security'
      />
    {{else if (eq error 'passwordRotateError')}}
      <FuiAuthError
        @errorText='To switch accounts,'
        @linkText='reset your password'
        @link='/account/personal/password'
      />
    {{/if}}

  <ul class='fui-account-switcher__list list--outdented list--unbulleted padding--xs'>
    {{#each accounts as |account|}}
      <li
        tabindex='0'
        role='button'
        aria-current={{if account.currentSession 'true' false}}
        aria-disabled={{if account.currentSession 'true' false}}
        class='fui-account-switcher__list-item fui-card fui-card--accentuated fui-card--focus-ring flex-row flex-row--vcenter padding--m'
        onclick={{action handleChooseAccount account}}
        onkeyup={{action handleKeyUp account}}
      >
        <div class='flex-col--1 text-align--left'>
          <p class='head-bold'>{{account.name}}</p>
          <p class='color--grey'>{{account.id}}</p>
        </div>
        {{#if account.defaultAccount}}
          <div class='font--italic margin-x--l'>Default</div>
        {{/if}}
        <div class="fui-account-switcher__action">
          {{#if account.currentSession}}
            <span class='fui-tag fui-tag--emerald'>Signed In</span>
          {{else}}
            <span class='flex-col--pushed fui-button--link button'>
              Sign In
            </span>
          {{/if}}
        </div>
      </li>
    {{else}}
      <li
        class='fui-account-switcher__list-item fui-account-switcher__list-item--fixed-height fui-card fui-skeleton-overlay'
      />
      <li
        class='fui-account-switcher__list-item fui-account-switcher__list-item--fixed-height fui-card fui-skeleton-overlay'
      />
      <li
        class='fui-account-switcher__list-item fui-account-switcher__list-item--fixed-height fui-card fui-skeleton-overlay'
      />
      <li
        class='fui-account-switcher__list-item fui-account-switcher__list-item--fixed-height fui-card fui-skeleton-overlay'
      />
    {{/each}}
  </ul>

  <p class='color--grey padding--s'>
    Don't see the account you're looking for?
    <ExternalLinkTo
      @text='Contact Support'
      @href='https://www.fastly.com/support'
    />.
  </p>
  `,
})
