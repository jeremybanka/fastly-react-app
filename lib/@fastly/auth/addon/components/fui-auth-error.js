import Component from '@ember/component'
import hbs from 'htmlbars-inline-precompile'

export default Component.extend({
  tagName: '',
  errorText: 'An error occured. Try again or ',
  linkText: 'Contact support',
  link: 'https://www.fastly.com/support',
  class: 'margin-bottom--m',
  dismissible: false,
  layout: hbs`
    <FuiMessage
      @type='error'
      @class='text-align--left {{class}}'
      @dismissible={{dismissible}}
    >
      <p role='alert' ...attributes >
        {{errorText}} <ExternalLinkTo
          @text={{linkText}}
          @href={{link}}
        />.
      </p>
    </FuiMessage>
  `,

  didInsertElement() {
    this._super(...arguments)
    const overlay = document.querySelector('.fui-auth-screen__overlay')
    if (overlay) {
      overlay.scrollTo({
        top: 0,
        behavior: 'smooth',
      })
    }
  },
})
