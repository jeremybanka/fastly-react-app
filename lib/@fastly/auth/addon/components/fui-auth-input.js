import Component from '@ember/component'
import { computed } from '@ember/object'
import { guidFor } from '@ember/object/internals'
import hbs from 'htmlbars-inline-precompile'

export default Component.extend({
  // Passed Properties
  // ---------------------------------------------------------------------------
  label: '',
  // Internal properties
  // ---------------------------------------------------------------------------
  tagName: '',
  inputId: computed(function () {
    return 'input-' + guidFor(this)
  }),

  layout: hbs`
    <div class='fui-auth-screen__input-group' aria-describedby={{tooltipId}}>
      <label class='fui-auth-screen__input-group-label flex-row margin-top--m margin-bottom--s' for={{inputId}}>
        {{label}}
        {{yield (hash contextual-link=(component 'yield'))}}
      </label>
      <Input @class='fui-input authentication__sign-in-input margin-top--s' @value={{value}} ...attributes id={{inputId}} />
    </div>
  `,
})
