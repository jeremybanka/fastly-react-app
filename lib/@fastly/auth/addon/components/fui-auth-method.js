import Component from '@ember/component'
import { computed } from '@ember/object'
import hbs from 'htmlbars-inline-precompile'
import { readOnly } from '@ember/object/computed'
import { inject as service } from '@ember/service'

export default Component.extend({
  tagName: '',
  session: service(),
  store: service(),

  accountName: computed('session.switchCustomerParams.customerId', function () {
    return this.store.peekRecord('account', this.session.switchCustomerParams.customerId).name
  }),

  ssoLoginUrl: readOnly('session.ssoLoginUrl'),
  layout: hbs`
    <h1 class="fui-auth-screen__header">Choose how to sign into <strong>{{accountName}}</strong></h1>
    <div class="padding-x--xxl padding-bottom--l">
      <ExternalLinkTo
        @href={{ssoLoginUrl}}
        class="button primary fui-login-form__button margin-bottom--l"
        aria-label="Sign in with single sign-on"
        target="_self"
      >
        Single sign-on
      </ExternalLinkTo>
      <LinkTo
        @route="switch-customer.login-required"
        class="button new fui-login-form__button margin-left--0"
        aria-label="Sign in with email and password"
      >
        Email and password
      </LinkTo>
    </div>
  `,
})
