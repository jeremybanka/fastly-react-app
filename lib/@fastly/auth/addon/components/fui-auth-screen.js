import Component from '@ember/component'
import hbs from 'htmlbars-inline-precompile'
import { inject as service } from '@ember/service'

export default Component.extend({
  // Services
  // ---------------------------------------------------------------------------
  router: service(),

  // Internal Properties
  // ---------------------------------------------------------------------------
  tagName: '',

  // Methods
  // ---------------------------------------------------------------------------
  cancel() {
    this.router.location.history.back()
  },

  // Template
  // ---------------------------------------------------------------------------

  layout: hbs`
    <div class='flex-row flex-row--hcenter text-align--center'>
      <div class='fui-card padding--l margin-top--xl'>
        <div class='fui-auth-screen__overlay'>
          <div class='fui-auth-screen__panel'>
            <FuiIcon @name='fastly-logo-red' @class='fastly-logo-red' />
            {{yield}}
            <button class='fui-auth-form__button cancel margin-top--m' {{action cancel}}>Cancel</button>
          </div>
        </div>
      </div>
    </div>
  `,
})
