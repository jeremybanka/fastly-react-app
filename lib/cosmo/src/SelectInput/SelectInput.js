/* @flow */

import * as React from "react";
import { FormattedMessage } from "react-intl";
import { Box, Icon, InputError, InputHelper, InputLabel, Text } from "../";
import styled from "styled-components";
import css from "@styled-system/css";

const StyledInputContainer = styled("div")((props) => {
  let borderColor = "grays.2";
  if (props.errored) borderColor = "reds.5";
  if (props.focused) borderColor = "grays.6";
  return css({
    position: "relative",
    height: "xl",
    borderRadius: "0",
    borderWidth: "1px",
    borderStyle: "solid",
    borderColor,
    "&:hover": {
      // scope to :enabled?
      // :focus should override :hover
      borderColor: "grays.3",
    },
    "&:focus": {
      borderColor: "grays.6",
    },
    backgroundColor: "grays.0",
  });
});

const StyledInput = styled("select")(() => {
  return css({
    outline: "none",
    display: "block",
    width: "100%",
    height: "100%",
    paddingLeft: "md",
    // | ------------------- |
    // | select text [arrow] |
    // | ------------------- |
    //              ▲   ▲   ▲
    //              A   B   C
    // internal padding (A) + icon width (B) + edge padding (C)
    //                          A          B             C
    //                          ▼          ▼             ▼
    // paddingRight: "calc(var(--space-sm) + 10px + var(--space-md))",
    color: "grays.6",
    background: "transparent",
    lineHeight: "normal",
    overflow: "hidden",
    cursor: "pointer",
    borderRadius: "0",
    "webkit-appearance": "none",
    "-moz-appearance": "none",
    "-ms-appearance": "none",
    "-o-appearance": "none",
    appearance: "none", // automatically prefixed?
    textIndent: "0.01px", // fixes a firefox bug i think
    textTransform: "none", // and on button ?
    border: "0",
    fontSize: "xs",
    fontWeight: "semibold",
    "&:disabled": {
      cursor: "not-allowed", // remove or apply to corresponding label?
      color: "grays.4",
      backgroundColor: "grays.1",
      // unset legacy overrides from Button.css
      opacity: "1",
      pointerEvents: "auto",
    },
  });
});

const StyledArrow = styled("span")(() => {
  return css({
    position: "absolute",
    top: "0",
    right: "7px" /* ArrowDropDown icon doesn't take up its full SVG viewbox */,
    bottom: "0",
    width: "30px",
    display: "flex",
    flexDirection: "column",
    justifyContent: "center",
    color: "grays.4",
    pointerEvents: "none",
  });
});

type TextType =
  | string
  | React.Element<typeof Text>
  | React.Element<typeof FormattedMessage>;

type Option = {
  value: string, // TextType?
  label: string,
  disabled?: boolean,
};

type FieldInputProps = {
  name: string,
  id: string,
  value: string,
  required?: boolean,
  disabled?: boolean,
  onChange: (e: any) => void,
  onFocus?: (e: any) => void,
  onBlur?: (e: any) => void,
};

type SelectInputProps = {
  input: FieldInputProps,
  label?: TextType,
  helper?: TextType,
  error?: ?TextType,
  options: Option[],
};

const SelectInput = (props: SelectInputProps) => {
  const { label, helper, error, input, options } = props;
  const { onFocus, onBlur, ...inputRest } = input;
  const [focus, setFocus] = React.useState(false);
  return (
    <Box>
      {/* if no label is provided, hide all metadata above the input */}
      {label ? (
        <Box marginBottom="xs">
          {/* $FlowFixMe - InputLabel expects text, not array */}
          <InputLabel htmlFor={input.id}>
            {label}
            {input.required ? null : (
              <Text color="grays.4" marginLeft="xs">
                (optional)
              </Text>
            )}
          </InputLabel>
        </Box>
      ) : null}
      <StyledInputContainer errored={!!error} focused={focus}>
        <StyledInput
          onFocus={(event) => {
            setFocus(true);
            if (input.onFocus) input.onFocus(event);
          }}
          onBlur={(event) => {
            setFocus(false);
            if (input.onBlur) input.onBlur(event);
          }}
          {...inputRest} // smelly
        >
          {options.map((option) => (
            <option
              key={option.value}
              value={option.value}
              disabled={option.disabled}
            >
              {option.label}
            </option>
          ))}
        </StyledInput>
        <StyledArrow>
          <Icon glyph="KeyboardArrowDown" />
        </StyledArrow>
      </StyledInputContainer>
      {helper ? (
        <Box marginTop="xs" fontSize="xxs">
          <InputHelper htmlFor={input.id}>{helper}</InputHelper>
        </Box>
      ) : null}
      {error ? (
        <Box marginTop="xs" fontSize="xxs">
          <InputError htmlFor={input.id}>{error}</InputError>
        </Box>
      ) : null}
    </Box>
  );
};

export default SelectInput;
