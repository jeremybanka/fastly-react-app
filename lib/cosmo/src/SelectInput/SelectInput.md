```jsx
<SelectInput
  input={{}}
  label="Choose"
  helper="Or we'll choose for you"
  options={[
    { value: "a", label: "A" },
    { value: "b", label: "B" },
    { value: "c", label: "C" },
  ]}
/>
```
