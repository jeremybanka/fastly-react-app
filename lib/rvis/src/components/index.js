/* @flow */

import AxisX from "./AxisX";
import AxisY from "./AxisY";
import Background from "./Background";
import Brush from "./Brush";
import BrushActions from "./BrushActions";
import BrushMask from "./BrushMask";
import Chart from "./Chart";
import ChartInteractions from "./ChartInteractions";
import ChartMarker from "./ChartMarker";
import ChartMarkerText from "./ChartMarkerText";
import LegendTotal from "./LegendTotal";
import Loading from "./Loading";

export {
  AxisX,
  AxisY,
  Background,
  Brush,
  BrushActions,
  BrushMask,
  Chart,
  ChartInteractions,
  ChartMarker,
  ChartMarkerText,
  LegendTotal,
  Loading,
};
