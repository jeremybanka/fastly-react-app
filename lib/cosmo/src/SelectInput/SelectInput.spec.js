/* @flow */

import React from "react";
import { render, shallow } from "enzyme";
import { InputLabel, InputHelper, InputError } from "components";
import SelectInput from "./SelectInput";

const handleBlur = jest.fn();
const handleChange = jest.fn();
const basicInputObject = {
  name: "tastes",
  id: "tastes-umami",
  checked: true,
  onChange: handleChange,
  onBlur: handleBlur,
};
const options = [
  { value: "a", label: "A" },
  { value: "b", label: "B" },
  { value: "c", label: "C" },
];

describe("SelectInput test suite", () => {
  it("renders the stored snapshot", () => {
    const component = (
      <SelectInput
        label="Umami"
        input={basicInputObject}
        options={options}
        helper="Mystery flavor"
      />
    );
    expect(render(component)).toMatchSnapshot();
  });

  it("renders with InputLabel", () => {
    const component = (
      <SelectInput label="Foo" input={basicInputObject} options={options} />
    );
    expect(shallow(component).find(InputLabel)).toHaveLength(1);
  });

  it("renders with InputHelper", () => {
    const component = (
      <SelectInput input={basicInputObject} options={options} helper="Bar" />
    );
    expect(shallow(component).find(InputHelper)).toHaveLength(1);
  });

  it("renders with InputError", () => {
    const component = (
      <SelectInput input={basicInputObject} options={options} error="Foo" />
    );
    expect(shallow(component).find(InputError)).toHaveLength(1);
  });

  it("renders without InputLabel, InputHelper, or InputError", () => {
    const component = (
      <SelectInput input={basicInputObject} options={options} />
    );
    expect(shallow(component).find(InputLabel)).toHaveLength(0);
    expect(shallow(component).find(InputHelper)).toHaveLength(0);
    expect(shallow(component).find(InputError)).toHaveLength(0);
  });
});
