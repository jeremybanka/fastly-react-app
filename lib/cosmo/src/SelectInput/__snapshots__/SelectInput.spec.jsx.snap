// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectInput test suite renders the stored snapshot 1`] = `
<div
  class="Box-sc-1v2etov-0 kAhoak"
>
  <div
    class="Box-sc-1v2etov-0 hKYyow"
  >
    <label
      class="Text-sc-6c1xd4-0 fktwGx"
      color="grays.6"
      font-size="xs"
      for="tastes-umami"
      letter-spacing="xs"
    >
      Umami
      <span
        class="Text-sc-6c1xd4-0 jWbYly"
        color="grays.4"
      >
        (optional)
      </span>
    </label>
  </div>
  <div
    class="SelectInput__StyledInputContainer-sc-1v1fvwr-0 eAgXAT"
  >
    <select
      checked=""
      class="SelectInput__StyledInput-sc-1v1fvwr-1 jQCMll"
      id="tastes-umami"
      name="tastes"
    >
      <option
        value="a"
      >
        A
      </option>
      <option
        value="b"
      >
        B
      </option>
      <option
        value="c"
      >
        C
      </option>
    </select>
    <span
      class="SelectInput__StyledArrow-sc-1v1fvwr-2 gXZxIC"
    >
      <span
        class="Icon"
      >
        <svg
          class="Icon-svg"
          style="fill:currentColor"
          viewBox="0 0 24 24"
        >
          <path
            d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
          />
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
        </svg>
      </span>
    </span>
  </div>
  <div
    class="Box-sc-1v2etov-0 dUYKhU"
    font-size="xxs"
  >
    <label
      class="Text-sc-6c1xd4-0 bDXzpq"
      color="grays.4"
      font-size="xxs"
      for="tastes-umami"
    >
      Mystery flavor
    </label>
  </div>
</div>
`;
